### ISSUE TITLE
Question

### ISSUE THREAD

**Issue by diogoazevedos** [fortunejs/fortune#75]
Is possible filter data, like great then, great and equal, between?
Thanks.

**Comment by daliwali** [comment]
Yes, by using the [`query` option](https://github.com/fortunejs/fortune/blob/master/lib/adapter/index.js#L68). If you want to use it, it is specific to the adapter and you may optionally extend a serializer for using it as a query parameter.

**Comment by diogoazevedos** [comment]
Hi Dali,

I am building a isomorphic webapp, and a need a prefix for fortune like **/api/** and create a route **/**. How can I do that using fortune? Is possible plug express in fortune, or fortune in express, or both in http?

Thanks.

**Comment by daliwali** [comment]
@diogoazevedos I think you asked another question and deleted it. there are 2 things you should do:

- use the `prefix` option in the serializer so that links are correct.
- mount fortune's request listener under a path. in express, it would be `app.use`.

**Comment by diogoazevedos** [comment]
@daliwali Yes, I tested `app.use` and just works. Thanks

**Comment by diogoazevedos** [comment]
Hi @0x8890,

How can I protect a resource with username and password?

Thanks.

**Comment by 0x8890** [comment]
To prevent a record from being returned, simply throw an error. Here's an example:

```js
import fortune from 'fortune'
import http from 'http'

const { errors } = fortune
const store = fortune.create()
store.defineType('post', {})
store.transformOutput('post', (context, record) => {
  if (context.request.meta['Authorization'] !== 'foo')
    throw new errors.UnauthorizedError(`The auth header must be "foo".`)
  return record
})
http.createServer(fortune.net.http(store)).listen(1337)
```

**Comment by diogoazevedos** [comment]
In a getting started example, when I try to log all metas, why is it empty?
Am I doing something wrong?

```shell
$ curl -X GET -H "Accept: application/vnd.api+json" \
-H "Content-Type: application/vnd.api+json" \
-H "Authorization: Basic ZGlvZ29hemV2ZWRvczoxMjM0NQ==" \
-v http://localhost:1337/users/1
```

```javascript
// [...]
store.transformOutput('user', (context, record) => {
  const { request: { meta } } = context
  console.log(meta) // {}
  if (meta.authorization !== 'Basic ZGlvZ29hemV2ZWRvczoxMjM0NQ==')
    throw new errors.UnauthorizedError('Authorization token doesn\'t match.')

  // Hide sensitive fields.
  delete record.salt
  delete record.key
  return record
})
// [...]
```

**Comment by 0x8890** [comment]
@diogoazevedos oops, my fault! somehow I forgot to assign request headers to the meta object, should be fixed in commit 64c744b1cc63f63d69d7a921f1110a718bc9b9a6.

**Comment by diogoazevedos** [comment]
Thanks @0x8890 :smile:
